packages/elements/src/configuration/elements/configuration.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  BasicElement,
+  TemplateResult,
+  html
+} from '@refinitiv-ui/core';
+import { customElement } from '@refinitiv-ui/core/decorators/custom-element.js';
+import { state } from '@refinitiv-ui/core/decorators/state.js';
+import { property } from '@refinitiv-ui/core/decorators/property.js';
+import { VERSION } from '../../version.js';
+import { efConfig, type Config as ConfigType, type ConfigIcon } from '../helpers/context.js';
+import { provide } from '@lit-labs/context';
+
+const DEFAULT_CONFIG: ConfigType = { icon: { map: {} } };
+
+@customElement('ef-configuration', { theme: false })
+export class Config extends BasicElement {
+
+  /**
+   * Element version number
+   * @returns version number
+   */
+  static get version (): string {
+    return VERSION;
+  }
+  
+  /**
+   * Render slot as light DOM
+   * @returns Element
+   */
+  protected createRenderRoot () {
+    return this;
+  }
+
+  /**
+   * Configuration data
+   * @ignore
+   */
+  @provide({ context: efConfig })
+  @state()
+  public config: ConfigType = DEFAULT_CONFIG;
+
+  @property({ type: Object })
+  set icon (icon: ConfigIcon) {
+    this.config = { ...this.config, icon };
+  }
```

Comment:
On this, I was made aware that we have predefined guidelines for the definition of public apis for components.
Please have a look at https://ui.refinitiv.com/custom-components/naming-conventions
We should align to those then

Reply:
Where are the predefined guidelines in the link? I have found nothing 😅

Reply:
The link contains naming conventions already in use throughout the library.
It happens that we already use a `config` property which may be suitable to expose here.
![Screenshot 2023-06-01 091605](https://github.com/Refinitiv/refinitiv-ui/assets/10496469/716f43dd-f02a-4dff-ab94-193daef850cc)

Reply:
So that means we could be setting configuration though out config and icon properties?

Reply:
We should pass the entire config to the component. I understand it is a bit verbose.
I see 2 options: use `config` with the whole config obj  (we could also provide to consumers a utility fn that they will use to define only what they need and the fn will fill the rest with defaults), or we can discuss with the team whether they are fine with this component api, even if it doesn't strictly adhere to conventions